### PR TITLE
[BEAM-4483]Add instructions about how to perform release validations

### PR DESCRIPTION
**Pull request by boyuanzz**
r: @aaltay @alanmyrvold
cc: @pabloem

### CODE REVIEW

File: src/contribute/release-guide.md [hunk 1]
```diff
@@ -511,7 +512,171 @@ If there are no issues, reply on the vote thread to close the voting. Then, tall
...
+    ```
+    Setup your service account as described in ```Java Mobile Game``` section above
+  
+    Produce data by using java injector
+    
+    ```
+    mvn archetype:generate \
+          -DarchetypeGroupId=org.apache.beam \
+          -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples \
+          -DarchetypeVersion=2.4.0 \
```
Comment (aaltay) [inline src/contribute/release-guide.md]: why 2.4.0 here?

File: src/contribute/release-guide.md [hunk 2]
```diff
@@ -511,7 +512,171 @@ If there are no issues, reply on the vote thread to close the voting. Then, tall
...
+          -DgroupId=org.example \
+          -DartifactId=word-count-beam \
+          -Dversion="0.1" \
+          -Dpackage=org.apache.beam.examples \
+          -DinteractiveMode=false
+          
+    mvn compile exec:java -Dexec.mainClass=org.apache.beam.examples.complete.game.injector.Injector \
+      -Dexec.args="${YOUR_PROJECT} ${YOUR_PUBSUB_TOPIC} none"
+    ```
+  * Run Leaderboard with Direct Runner
```
Comment (aaltay) [inline src/contribute/release-guide.md]: Should we talk about how to check that results are actually produced and correct?

**Comment by yifanzou** [comment]
We have some cases already automated. Shall we teach validators how to run it instead of verify examples one by one. Might make their life easier?

File: src/contribute/release-guide.md (hunk 1 above)
Comment (boyuanzz) [reply src/contribute/release-guide.md]: We just use java injector here, so we can get jar from last release.

File: src/contribute/release-guide.md (hunk 2 above)
Comment (boyuanzz) [reply src/contribute/release-guide.md]: right, I'll add how to verify results. Thanks for pointing it out!

**Comment by boyuanzz** [comment]
Re: @yifanzou 
For the tests which has been automated, I put automation test command in the guide. For the rest of the tests, I list manually run commands.

File: src/contribute/release-guide.md (hunk 2 above)
Comment (boyuanzz) [reply src/contribute/release-guide.md]: Commit https://github.com/apache/beam-site/pull/468/commits/08ddcca4c3b1ec0c1983bf249ac3fcc4984ca6ac added more instructions about how to verify outputs.

File: src/contribute/release-guide.md (hunk 1 above)
Comment (alanmyrvold) [reply src/contribute/release-guide.md]: We should generate it from the release being validated, so 2.5.0 now, 2.6.0 for the next release

File: src/contribute/release-guide.md [hunk 3]
```diff
@@ -511,7 +512,183 @@ If there are no issues, reply on the vote thread to close the voting. Then, tall
...
+    
+### Run validation tests
+All tests listed in this [spreadsheet](https://docs.google.com/spreadsheets/d/1qk-N5vjXvbcEk68GjbkSZTR8AGqyNUM-oLFo_ZXBpJw/edit#gid=152451807)
+
+* Java Quickstart Validation
 
+  Direct Runner:
+  ```
+  ./gradlew :beam-runners-direct-java:runQuickstartJavaDirect \ 
+  -Prepourl=https://repository.apache.org/content/repositories/orgapachebeam-1041 \
```
Comment (alanmyrvold) [inline src/contribute/release-guide.md]: Need to add a comment to replace 1041 by the version being used for the release

File: src/contribute/release-guide.md (hunk 1 above)
Comment (boyuanzz) [reply src/contribute/release-guide.md]: It seems like, we don't upload artifacts into maven central  during vote process. So we cannot get release version artifacts using this cmd before it releases.

**Comment by boyuanzz** [comment]
Hey @alanmyrvold , please review commit https://github.com/apache/beam-site/pull/468/commits/7792feeb9b74fec37449c4b7ba831393d17f066c . Thanks for your help!

File: src/contribute/release-guide.md (hunk 1 above)
Comment (lukecwik) [reply src/contribute/release-guide.md]: You can get the artifacts from the staging repo, you need to configure an archetype repo in your `~/.m2/settings.xml` with an archetype repo like:
```
<settings>
  <profiles>
   <profile>
     <id>release-repo</id>
     <activation>
       <activeByDefault>true</activeByDefault>
     </activation>
     <repositories>
       <repository>
         <id>Release 2.4.0 RC3</id>
         <name>Release 2.4.0 RC3</name>
         <url>https://repository.apache.org/content/repositories/orgapachebeam-1031/</url>
       </repository>
     </repositories>
   </profile>
  </profiles>
</settings>
```

and specify `-DarchetypeCatalog=internal`

Comment (boyuanzz) [reply src/contribute/release-guide.md]: I added more instructions in commit https://github.com/apache/beam-site/pull/468/commits/9b0ec38866cdd67a2eba415ae318f3872011e723. @lukecwik  @aaltay

File: src/contribute/release-guide.md [hunk 4]
```diff
@@ -511,7 +512,208 @@ If there are no issues, reply on the vote thread to close the voting. Then, tall
...
+  * Run Leaderboard with Direct Runner
+    ```
+    python -m apache_beam.examples.complete.game.leader_board \
+    --project=${YOUR_PROJECT} \
+    --topic projects/${YOUR_PROJECT}/topics/${YOUR_PUBSUB_TOPIC} \
+    --dataset ${USER}_test
+    ```
+    Inspect results:
+    * Check whether there is any error messages in console.
+    * Goto your BigQuery console and check whether your ${USER}_test has leader_board table.
```
Comment (aaltay) [inline src/contribute/release-guide.md]: I think both leader board and game stats should produce 2 tables: 1 for user and 1 for team. Could you mention that.

Should we include any example rows to look for inside the tables?

Comment (boyuanzz) [reply src/contribute/release-guide.md]: Please review the latest commit https://github.com/apache/beam-site/pull/468/commits/3a1852ab0a9913cda76e246a6c9ffcb4b4728e8f .

File: src/contribute/release-guide.md [hunk 5]
```diff
@@ -511,7 +512,220 @@ If there are no issues, reply on the vote thread to close the voting. Then, tall
     There are no disapproving votes.
 
     Thanks everyone!
+    
+### Run validation tests
+All tests listed in this [spreadsheet](https://docs.google.com/spreadsheets/d/1qk-N5vjXvbcEk68GjbkSZTR8AGqyNUM-oLFo_ZXBpJw/edit#gid=152451807)
```
Comment (alanmyrvold) [inline src/contribute/release-guide.md]: Change to https://s.apache.org/beam-release-validation ?

Comment (boyuanzz) [reply src/contribute/release-guide.md]: Fixed in https://github.com/apache/beam-site/pull/468/commits/ce2286c3569959ea1f3821bf8522127ed7c73bc4 .

**Comment by aaltay** [comment]
@asfgit merge

File: src/contribute/release-guide.md [hunk 6]
```diff
@@ -511,7 +512,220 @@ If there are no issues, reply on the vote thread to close the voting. Then, tall
...
+    --fixed_window_duration ${SOME_SMALL_DURATION}
+    ```
+    Inspect results:
+    * Goto your Dataflow job console and check whether there is any error.
+    * Goto your BigQuery console and check whether your ${USER}_test has game_stats_teams and game_stats_sessions table.
+    * bq head -n 10 ${USER}_test.game_stats_teams
+    * bq head -n 10 ${USER}_test.game_stats_sessions
 ### Checklist to proceed to the finalization step
 
 1. Community votes to release the proposed candidate, with at least three approving PMC votes
```
Comment (alanmyrvold) [inline src/contribute/release-guide.md]: All community members are encouraged to vote, with +1 to approve the release and -1 if there is an issue with the release. If there is an issue, clarify in the email. Only PMC members can cast binding votes.

**Pull request closed without merging**